File: python/78.py
class Solution(object):
    def search(self,nums,result,index):
        if index == len(nums):
            self.results.append(result)
            return
        self.search(nums,result +[nums[index]],index+1)
        self.search(nums,result,index+1)
        
    def subsets(self, nums):
        self.results = []
        self.search(sorted(nums),[],0)
        return self.results

File: c++/78.cpp
#include <vector>

class Solution {
public: 
    std::vector<std::vector<int>> subsets(std::vector<int>& nums ) {
        std::vector<std::vector<int>> result;
        std::vector<int> item;
        result.push_back(item);
        generate(0,nums,item,result);
        return result;
    }
private:
    void generate(int i, std::vector<int>& nums,
    std::vector<int> & item, std::vector<std::vector<int>> &result) {
        if (i>= nums.size()) {
            return;
        }
        item.push_back(nums[i]);
        result.push_back(item);
        generate(i+1,nums,item,result);
        item.pop_back();
        generate(i+1,nums,item,result);
    }
}